Validation/HGCalValidation/plugins/HGCalWaferValidation.cc

```diff
@@ -213,36 +233,54 @@ void HGCalWaferValidation::ProcessWaferLayer(DDCompactView::GraphWalker& walker)
       //edm::LogVerbatim(logcat) << "rotStr " << rotStr << " rotCode " << rotCode;
 
       // convert shape code to wafer types defined in HGCalTypes.h
-      waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferOut;
-      if (shapeStr == "F")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferFull;
-      else if (shapeStr == "a")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferHalf;
-      else if (shapeStr == "am")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferHalf2;
-      else if (shapeStr == "b")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferFive;
-      else if (shapeStr == "bm")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferFive2;
-      else if (shapeStr == "c")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferThree;
-      else if (shapeStr == "d")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferSemi;
-      else if (shapeStr == "dm")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferSemi2;
-      else if (shapeStr == "g")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferChopTwo;
-      else if (shapeStr == "gm")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferChopTwoM;
+      waferInfo.shapeCode = shapeStr == "F"    ? HGCalTypes::WaferPartialType::WaferFull
+                            : shapeStr == "a"  ? HGCalTypes::WaferPartialType::WaferHalf
+                            : shapeStr == "am" ? HGCalTypes::WaferPartialType::WaferHalf2
+                            : shapeStr == "b"  ? HGCalTypes::WaferPartialType::WaferFive
+                            : shapeStr == "bm" ? HGCalTypes::WaferPartialType::WaferFive2
+                            : shapeStr == "c"  ? HGCalTypes::WaferPartialType::WaferThree
```

Comment:
I am not sure if this kind of structures are oreferred over else ifs, I let experts to comment
@perrotta @qliphy

Reply:
I prefer this way because:
- Makes things more readable, in table-like form
- Allows assignment to a `const`, which makes things more robust

Reply:
Surely the changed code is more readable

Reply:
I kind of agree with the comment from @jfernan2 
From the performance point of view, there's no gain.
From the readability, the `if`-`else` is definitely more readable.
On a related note, instead of having this huge cascade of cases, would it make sense to have a map that, given the `shapeStr` will return the correct `WaferPartialType`? This would make the code even more readable.

Reply:
You mean define a constant `std::map` within this module?

Then we would need 3 maps for that. One for old format mapping and DD, one for newer format low-density mapping and another one for newer format high-density mapping.

Reply:
Yes, a `std::map` would do the job perfectly.
Keep in mind that the handling of the old format is just temporary and, as soon as we converge on something close to final, we can simply drop the rest.

Thanks!

Reply:
Thanks! Will implement this now.